components/pthread/include/esp_pthread.h

```diff
@@ -21,6 +21,7 @@ extern "C" {
 /** pthread configuration structure that influences pthread creation */
 typedef struct {
     size_t stack_size;  ///< The stack size of the pthread
+    uint32_t stack_alloc_caps;   ///< A bit mask of memory capabilities (MALLOC_CAPS*) to use when allocating the stack. If zero, MALLOC_CAP_DEFAULT is assumed.
```

Comment:
I'm afraid this might cause trouble for users who initialized each member of `esp_pthread_cfg_t` manually like this:
```
    esp_pthread_cfg_t cfg;
    cfg.thread_name = name;
    cfg.pin_to_core = core_id;
    cfg.stack_size = stack;
    cfg.prio = prio;
    cfg.inherit_cfg = false;
``` 
If they would upgrade IDF including this change, `esp_pthread_cfg_t.stack_alloc_caps` is uninitialized. Once the configuration is used, this would produce a crash at runtime at best or a way more subtle memory issue in the worst case.

I don't have a definitive solution for this yet. The `esp_pthread_set_cfg` makes it hard to introduce non-breaking configuration changes.

Reply:
Sorry, I missed that our documentation is [explicitly advising users to zero-initialize configuration structures](https://docs.espressif.com/projects/esp-idf/en/release-v5.0/esp32/api-reference/api-conventions.html). You can consider this comment resolved, then.